designs/2019-ignore-passed-file-paths/README.md

```diff
@@ -0,0 +1,52 @@
+-   Start Date: 2019-09-22
+-   RFC PR: _not yet created_
+-   Authors: Fabian (fabb)
+
+# Add CLI Option to Ignore Passed File Paths
+
+## Summary
+
+A new CLI option should be added that causes ESLint to respect the ignore patterns in `.eslintignore` even for file paths that are passed directly to the CLI.
+
+## Motivation
+
+Tools like [lint-staged](https://github.com/okonet/lint-staged) allow to run ESLint and other CLI tools on each commit on all changed files by passing the file paths directly as parameters to the tools. For example in the case the files `file1.js` and `file2.js` are changed, lint-staged will call `eslint —fix file1.js file2.js`.
+
+The problem is that the git repo might contain files that should be ignored by ESLint via `.eslintignore`. In case such a file is changed and committed, lint-staged will pass its path to the ESLint CLI nonetheless. ESLint currently works in a way where it ignores `.eslintignore` for files explicitly passed to the ESLint CLI. That causes the file to be linted, and in case of lint-staged, will prevent the git commit to succeed if errors are found.
+
+## Detailed Design
+
+A new CLI option `--force-ignore` should be added to the ESLint CLI. When this option is given, ESLint should ignore files according to `.eslintignore` even when files are passed to the ESLint CLI explicitly.
```

Comment:
We need to describe the updates of `CLIEngine` class because `eslint` CLI command uses the `CLIEngine` class and it's a public API.

The `CLIEngine` class has constructor options that correspond to CLI options. This RFC should add an option for the new `--force-ignore`.